### DISCUSSION TITLE
`static` vs `.with_state` for shared state

### DISCUSSION THREAD

**Discussion by jjangga0214** [tokio-rs/axum#2269] [General]
The [docs](https://github.com/tokio-rs/axum/blob/dea36db400f27c025b646e5720b9a6784ea4db6e/examples/key-value-store/src/main.rs) shows examples of `.with_state`.

But we may be able to use a `static` variable for shared state.

Is there a benefit of using `.with_state`?

Thanks.

**Comment by ttys3** [comment]
depends on a global static var will make your code hard to understand and to test.
